client/src/components/SelectionDialog/SelectionCard.vue

```diff
@@ -0,0 +1,224 @@
+                    class="cursor-pointer align-self-end"
+                    :checked="checked"
+                    @change="onSelect(props.item)" />
+            </div>
+
+            <div class="selection-card-data" @click="onSelect(props.item)">
+                <div class="selection-card-body">
+                    <span>
+                        <pre v-if="isEncoded" :title="`label-${props.item.url}`">
+                                <code>{{ props.item.label }}</code>
+                            </pre>
+                        <span v-else-if="props.item.isLeaf">
+                            <FontAwesomeIcon :icon="props.leafIcon" fixed-width size="sm" />
+                            <span :title="`label-${props.item.url}`">
+                                {{ props.item.label }}
+                            </span>
+                        </span>
+                        <span v-else :title="`label-${props.item.url}`" @click="onOpen(props.item)">
+                            <FontAwesomeIcon :icon="props.folderIcon" fixed-width size="sm" />
+
+                            <span class="parent-title">
+                                {{ props.item.label }}
+                            </span>
+                        </span>
+                    </span>
+
+                    <span v-if="props.item.update_time || props.item.time" class="selection-card-header-right">
+                        <BBadge
+                            v-if="!props.item.isLeaf"
+                            v-b-tooltip.hover.noninteractive
+                            :title="parentSizeTitle"
+                            class="outline-badge cursor-pointer">
+                            <FontAwesomeIcon :icon="faDatabase" fixed-width />
+                            {{ props.item.size || 0 }}
+                        </BBadge>
+
+                        {{ formatTime(props.item.update_time ?? props.item.time) }}
+                    </span>
+                </div>
+
+                <div class="selection-card-footer">
+                    <StatelessTags
+                        v-if="(props.item.tags ?? []).length > 0"
+                        :value="props.item.tags"
+                        :disabled="true" />
+
+                    <div class="selection-card-description" :title="`details-${props.item.url}`">
+                        {{ props.item.details }}
+                    </div>
+
+                    <BBadge
+                        v-if="props.item.isLeaf"
+                        v-b-tooltip.hover.noninteractive
+                        :title="leafSizeTitle"
+                        class="outline-badge cursor-pointer">
+                        <FontAwesomeIcon :icon="faDatabase" fixed-width />
+                        {{ bytesToString(props.item.size || 0) }}
+                    </BBadge>
+                </div>
+            </div>
+        </div>
+    </div>
+</template>
+
+<style scoped lang="scss">
```

Comment:
This is yet another card with custom styles and colors. There should be no custom color in the client code.